core/src/main/java/software/amazon/smithy/java/core/serde/BufferingFlatMapProcessor.java

```diff
@@ -64,13 +64,23 @@ public final void subscribe(Flow.Subscriber<? super O> subscriber) {
 
     @Override
     public final void onNext(I item) {
+        enqueueItem(item);
+        flush();
+    }
+
+    /**
+     * Adds the item to the queue for later delivery. The queue will get flushed upon calls to {@link #request} or
+     * {@link #onNext(Object)}.
+     *
+     * @param item The item to add to the queue
+     */
+    public final void enqueueItem(I item) {
```

Comment:
We can not add a new API that must be used for proper flow outside of the rx spec